File: game-service/src/main/java/com/insecurebank/game/util/HashUtil.java
package com.insecurebank.game.util;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

/**
 * SHA1 without salt -- matches the bank's own (intentionally weak) password
 * hashing scheme in auth-service, since we're validating against the same
 * customers.password column.
 */
public final class HashUtil {

    private HashUtil() {
    }

    public static String sha1Hash(String input) {
        try {
            MessageDigest digest = MessageDigest.getInstance("SHA-1");
            byte[] hashBytes = digest.digest(input.getBytes());
            StringBuilder sb = new StringBuilder();
            for (byte b : hashBytes) {
                sb.append(String.format("%02x", b));
            }
            return sb.toString();
        } catch (NoSuchAlgorithmException e) {
            throw new RuntimeException(e);
        }
    }
}

File: auth-service/src/main/kotlin/com/insecurebank/auth/util/HashUtil.kt
package com.insecurebank.auth.util

import java.security.MessageDigest

object HashUtil {
    
    /**
     * Hash a string using SHA1 without salt (intentionally vulnerable)
     * This is insecure and should never be used in production!
     */
    fun sha1Hash(input: String): String {
        val digest = MessageDigest.getInstance("SHA-1")
        val hashBytes = digest.digest(input.toByteArray())
        return hashBytes.joinToString("") { "%02x".format(it) }
    }
    
    /**
     * Verify if plain text matches SHA1 hash
     */
    fun verifySha1Hash(plainText: String, hash: String): Boolean {
        return sha1Hash(plainText) == hash
    }
}